crashes when attemting to use any devices

whenever i place a block then right click to use it it crashes minecraft.

im using rftools-2.42 
minecraft crashes when i select a device e.g.sheild generateor whilst holding another object it doesn't crash if i have nothing held however




---- Minecraft Crash Report ----
// You're mean.

Time: 19/02/15 19:00
Description: Unexpected error

java.lang.NoClassDefFoundError: cofh/api/item/IToolHammer
	at com.mcjty.container.GenericBlock.testWrenchUsage(GenericBlock.java:125)
	at com.mcjty.rftools.blocks.shield.ShieldBlock.func_149727_a(ShieldBlock.java:76)
	at net.minecraft.client.multiplayer.PlayerControllerMP.func_78760_a(PlayerControllerMP.java:326)
	at net.minecraft.client.Minecraft.func_147121_ag(Minecraft.java:1436)
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1942)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:962)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:887)
	at net.minecraft.client.main.Main.main(SourceFile:148)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
Caused by: java.lang.ClassNotFoundException: cofh.api.item.IToolHammer
	at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:101)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	... 14 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Stacktrace:
	at com.mcjty.container.GenericBlock.testWrenchUsage(GenericBlock.java:125)
	at com.mcjty.rftools.blocks.shield.ShieldBlock.func_149727_a(ShieldBlock.java:76)
	at net.minecraft.client.multiplayer.PlayerControllerMP.func_78760_a(PlayerControllerMP.java:326)
	at net.minecraft.client.Minecraft.func_147121_ag(Minecraft.java:1436)

-- Affected level --
Details:
	Level name: MpServer
	All players: 1 total; [EntityClientPlayerMP['Ruadri'/211, l='MpServer', x=-92.40, y=73.62, z=1.05]]
	Chunk stats: MultiplayerChunkCache: 225, 225
	Level seed: 0
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: 
	Level spawn location: World: (-27,64,21), Chunk: (at 5,4,5 in -2,1; contains blocks -32,0,16 to -17,255,31), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
	Level time: 910579 game time, 921861 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: false
	Forced entities: 32 total; [EntityBlock['unknown'/14592, l='MpServer', x=2147483647.00, y=2147483647.00, z=2147483647.00], EntityBlock['unknown'/14593, l='MpServer', x=2147483647.00, y=2147483647.00, z=2147483647.00], EntityBlock['unknown'/14594, l='MpServer', x=2147483647.00, y=2147483647.00, z=2147483647.00], EntityBlock['unknown'/14595, l='MpServer', x=2147483647.00, y=2147483647.00, z=2147483647.00], EntityBlock['unknown'/14596, l='MpServer', x=2147483647.00, y=2147483647.00, z=2147483647.00], EntityBlock['unknown'/14597, l='MpServer', x=2147483647.00, y=2147483647.00, z=2147483647.00], EntityGate['entity.AncientWarfareStructure.aw_gate.name'/134, l='MpServer', x=-73.50, y=104.00, z=-2.50], EntityBlock['unknown'/14598, l='MpServer', x=2147483647.00, y=2147483647.00, z=2147483647.00], EntityGate['entity.AncientWarfareStructure.aw_gate.name'/135, l='MpServer', x=-67.50, y=104.00, z=3.50], EntityBlock['unknown'/14599, l='MpServer', x=2147483647.00, y=2147483647.00, z=2147483647.00], EntityGate['entity.AncientWarfareStructure.aw_gate.name'/136, l='MpServer', x=-73.50, y=104.00, z=9.50], EntityBlock['unknown'/14600, l='MpServer', x=2147483647.00, y=2147483647.00, z=2147483647.00], EntityGate['entity.AncientWarfareStructure.aw_gate.name'/137, l='MpServer', x=-79.50, y=104.00, z=3.50], EntityBlock['unknown'/14601, l='MpServer', x=2147483647.00, y=2147483647.00, z=2147483647.00], EntityGate['entity.AncientWarfareStructure.aw_gate.name'/138, l='MpServer', x=-73.50, y=112.00, z=9.50], EntityBlock['unknown'/14602, l='MpServer', x=2147483647.00, y=2147483647.00, z=2147483647.00], EntityBlock['unknown'/14603, l='MpServer', x=2147483647.00, y=2147483647.00, z=2147483647.00], EntityBat['Bat'/17950, l='MpServer', x=-74.22, y=27.84, z=54.22], EntityItem['item.item.egg'/17183, l='MpServer', x=-129.56, y=65.13, z=-46.25], EntityBat['Bat'/17951, l='MpServer', x=-80.28, y=28.59, z=57.31], EntityBat['Bat'/17952, l='MpServer', x=-73.13, y=29.81, z=58.25], EntityClientPlayerMP['Ruadri'/211, l='MpServer', x=-92.40, y=73.62, z=1.05], EntityPig['Pig'/104, l='MpServer', x=-153.19, y=69.00, z=-70.88], EntityChicken['Chicken'/105, l='MpServer', x=-157.41, y=65.00, z=-57.59], EntityTrail['unknown'/236, l='MpServer', x=-92.40, y=73.62, z=1.05], EntityPig['Pig'/112, l='MpServer', x=-142.47, y=65.00, z=-64.25], EntityChicken['Chicken'/113, l='MpServer', x=-138.41, y=66.00, z=-64.53], EntityItem['item.item.egg'/17074, l='MpServer', x=-155.03, y=65.13, z=-58.31], EntityPig['Pig'/115, l='MpServer', x=-134.97, y=64.00, z=-60.69], EntityPig['Pig'/116, l='MpServer', x=-133.16, y=64.00, z=-61.53], EntityChicken['Chicken'/117, l='MpServer', x=-130.41, y=65.00, z=-46.47], NpcVikingArcher['Viking Archer'/16185, l='MpServer', x=-207.31, y=65.00, z=-69.69]]
	Retry entities: 0 total; []
	Server brand: fml,forge
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.client.multiplayer.WorldClient.func_72914_a(WorldClient.java:373)
	at net.minecraft.client.Minecraft.func_71396_d(Minecraft.java:2433)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:916)
	at net.minecraft.client.main.Main.main(SourceFile:148)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
	Minecraft Version: 1.7.10
	Operating System: Windows 8.1 (x86) version 6.3
	Java Version: 1.8.0_25, Oracle Corporation
	Java VM Version: Java HotSpot(TM) Client VM (mixed mode), Oracle Corporation
	Memory: 132398600 bytes (126 MB) / 523501568 bytes (499 MB) up to 523501568 bytes (499 MB)
	JVM Flags: 6 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx512M -XX:+UseConcMarkSweepGC -XX:+CMSIncrementalMode -XX:-UseAdaptiveSizePolicy -Xmn128M
	AABB Pool Size: 0 (0 bytes; 0 MB) allocated, 0 (0 bytes; 0 MB) used
	IntCache: cache: 0, tcache: 0, allocated: 12, tallocated: 96
	FML: MCP v9.05 FML v7.10.85.1230 Minecraft Forge 10.13.2.1230 51 mods loaded, 51 mods active
	mcp{9.05} [Minecraft Coder Pack] (minecraft.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	FML{7.10.85.1230} [Forge Mod Loader] (forge-1.7.10-10.13.2.1230.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	Forge{10.13.2.1230} [Minecraft Forge] (forge-1.7.10-10.13.2.1230.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	LanteaCraft-Core{1.0} [LanteaCraft Core] (minecraft.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	OpenComputers|Core{1.0.0} [OpenComputers (Core)] (minecraft.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	TooManyItems{1.7.10} [TooManyItems] (minecraft.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	BuildCraft|Core{6.2.6} [BuildCraft] (buildcraft-6.2.6.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	AncientWarfare{2.4.83-beta-MC1.7.10} [Ancient Warfare Core] (ancientwarfare-2.4.83-beta-MC1.7.10.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	AncientWarfareAutomation{2.4.83-beta-MC1.7.10} [Ancient Warfare Automation] (ancientwarfare-2.4.83-beta-MC1.7.10.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	AncientWarfareNEIPlugin{2.4.83-beta-MC1.7.10} [Ancient Warfare NEI Plugin] (ancientwarfare-2.4.83-beta-MC1.7.10.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	AncientWarfareNpc{2.4.83-beta-MC1.7.10} [Ancient Warfare NPCs] (ancientwarfare-2.4.83-beta-MC1.7.10.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	AncientWarfareStructure{2.4.83-beta-MC1.7.10} [Ancient Warfare Structures] (ancientwarfare-2.4.83-beta-MC1.7.10.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	AncientWarfareVehicle{2.4.83-beta-MC1.7.10} [Ancient Warfare Vehicles] (ancientwarfare-2.4.83-beta-MC1.7.10.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	examplemod{1.0} [Example Mod] (Avatar TLB 0.7.4.jar(1).zip) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	kieranvs_avatar{0.7.4} [Avatar] (Avatar TLB 0.7.4.jar(1).zip) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	bdlib{1.6.2.41} [BD Lib] (bdlib-mc1710-1.6.2.41.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	AWWayofTime{v1.3.0} [Blood Magic: Alchemical Wizardry] (BloodMagic-1.7.10-1.3.0-1.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	BuildCraft|Transport{6.2.6} [BC Transport] (buildcraft-6.2.6.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	BuildCraft|Silicon{6.2.6} [BC Silicon] (buildcraft-6.2.6.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	BuildCraft|Builders{6.2.6} [BC Builders] (buildcraft-6.2.6.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	BuildCraft|Energy{6.2.6} [BC Energy] (buildcraft-6.2.6.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	BuildCraft|Factory{6.2.6} [BC Factory] (buildcraft-6.2.6.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	buildhelper{1.0.0} [Build Helper] (BuildHelper_v1.0.0.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	craftguide{1.6.8.1} [CraftGuide] (CraftGuide-1.6.8.1.zip) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	Baubles{1.0.1.8} [Baubles] (Baubles-1.7.10-1.0.1.8.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	evilcraft{0.7.2b} [EvilCraft] (EvilCraft-1.7.10-0.7.2b.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	forcefieldmod{1.0.0.1} [Force Field Mod] (forcefield-172-1011.zip) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	advgenerators{0.9.8.38} [Advanced Generators] (generators-mc1710-0.9.8.38.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	Thaumcraft{4.2.1.4} [Thaumcraft] (Thaumcraft-1.7.10-4.2.1.4.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	GraveStone{2.11.3} [GraveStone] (GraveStone-2.11.3.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	iChunUtil{4.0.0} [iChunUtil] (iChun-Util-Mod-1.7.10.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	instantblocks{1.5.4} [Instant Blocks] (InstantBlocks_v1.5.4.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	ironman{1.0} [The Iron Man mod] (IronMan-1.7.2-1.0.6.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	ForgeMultipart{1.1.0.314} [Forge Multipart] (ForgeMultipart-1.7.10-1.1.0.314-universal.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	OpenComputers{1.4.9.38} [OpenComputers] (OpenComputers-MC1.7.10-1.4.9.38-universal(1).jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	LanteaCraft{1.0.0-68} [LanteaCraft] (LanteaCraft-1.7.10-68.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	Magitek{0.2.0} [Magitek] (Magitek-1.7.10-0.2.0.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	Mekanism{7.1.1} [Mekanism] (Mekanism-1.7.10-7.1.1.127.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	MekanismGenerators{7.1.1} [MekanismGenerators] (MekanismGenerators-1.7.10-7.1.1.127.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	meteors{2.13.1} [Falling Meteors] (meteors-1.7.10-2.13.1.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	Morph{0.9.1} [Morph] (Morph-Beta-0.9.1.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	clearingblock{0.3} [ClearingBlock Mod] (Nuclear Bomb Mod 1.0v.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	origin{3.4.0} [Origin] (Origin-1.7.10-3.4.0.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	rftools{2.42} [RFTools] (rftools-2.42(1).jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	sgs_salandaars_orb{1.0} [SGS Salandaar's Orb 1.7.2] (SalandaarsOrb-1.7.2-1.0.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	Sanguimancy{1.1.8} [Sanguimancy] (Sanguimancy-1.7.10-1.1.8-4.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	SGCraft{1.8.4} [Greg's SG Craft] (SGCraft-1.8.4-mc1.7.10.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	tpmod{1.9.0} [Teleportation Mod] (Teleportation Mod 1.9 for MC 1.7.10.zip) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	witchery{0.22.0} [Witchery] (witchery-1.7.10-0.22.0.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	McMultipart{1.1.0.314} [Minecraft Multipart Plugin] (ForgeMultipart-1.7.10-1.1.0.314-universal.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	ForgeMicroblock{1.1.0.314} [Forge Microblocks] (ForgeMultipart-1.7.10-1.1.0.314-universal.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized->Available->Available->Available->Available
	Launched Version: 1.7.10-Forge10.13.2.1230
	LWJGL: 2.9.1
	OpenGL: Intel(R) HD Graphics 4000 GL version 4.0.0 - Build 10.18.10.3621, Intel
	GL Caps: Using GL 1.3 multitexturing.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Anisotropic filtering is supported and maximum anisotropy is 16.
Shaders are available because OpenGL 2.1 is supported.

	Is Modded: Definitely; Client brand changed to 'fml,forge'
	Type: Client (map_client.txt)
	Resource Packs: []
	Current Language: English (US)
	Profiler Position: N/A (disabled)
	Vec3 Pool Size: 0 (0 bytes; 0 MB) allocated, 0 (0 bytes; 0 MB) used
	Anisotropic Filtering: Off (1)

--- a/src/main/java/com/mcjty/rftools/RFTools.java
+++ b/src/main/java/com/mcjty/rftools/RFTools.java
@@ -37,10 +37,12 @@ import org.apache.logging.log4j.Logger;
 
 import java.util.EnumMap;
 
-@Mod(modid = RFTools.MODID, version = RFTools.VERSION)
+@Mod(modid = RFTools.MODID, name="RFTools", dependencies = "required-after:Forge@["+RFTools.MIN_FORGE_VER+",);required-after:CoFHCore@["+RFTools.MIN_COFHCORE_VER+",)", version = RFTools.VERSION)
 public class RFTools {
     public static final String MODID = "rftools";
     public static final String VERSION = "2.50";
+    public static final String MIN_FORGE_VER = "10.13.0.1230";
+    public static final String MIN_COFHCORE_VER = "1.7.10R3.0.0B9";
 
     @SidedProxy(clientSide="com.mcjty.rftools.ClientProxy", serverSide="com.mcjty.rftools.ServerProxy")
     public static CommonProxy proxy;

